src/Common/src/Interop/Unix/Interop.Errors.cs

```diff
@@ -1,48 +1,166 @@
+        ENOMSG           = 0x10032,           // No message of the desired type.
+        ENOPROTOOPT      = 0x10033,           // Protocol not available.
+        ENOSPC           = 0x10034,           // No space left on device.
+        ENOSR            = 0x10035,           // No STREAM resources.
+        ENOSTR           = 0x10036,           // Not a STREAM.
+        ENOSYS           = 0x10037,           // Function not supported.
+        ENOTCONN         = 0x10038,           // The socket is not connected.
+        ENOTDIR          = 0x10039,           // Not a directory or a symbolic link to a directory.
+        ENOTEMPTY        = 0x1003A,           // Directory not empty.
+        ENOTRECOVERABLE  = 0x1003B,           // State not recoverable.
+        ENOTSOCK         = 0x1003C,           // Not a socket.
+        ENOTSUP          = 0x1003D,           // Not supported (same value as EOPNOTSUP).
+        ENOTTY           = 0x1003E,           // Inappropriate I/O control operation.
+        ENXIO            = 0x1003F,           // No such device or address.
+        EOVERFLOW        = 0x10040,           // Value too large to be stored in data type.
+        EOWNERDEAD       = 0x10041,           // Previous owner died.
+        EPERM            = 0x10042,           // Operation not permitted.
+        EPIPE            = 0x10043,           // Broken pipe.
+        EPROTO           = 0x10044,           // Protocol error.
+        EPROTONOSUPPORT  = 0x10045,           // Protocol not supported.
+        EPROTOTYPE       = 0x10046,           // Protocol wrong type for socket.
+        ERANGE           = 0x10047,           // Result too large.
+        EROFS            = 0x10048,           // Read-only file system.
+        ESPIPE           = 0x10049,           // Invalid seek.
+        ESRCH            = 0x1004A,           // No such process.
+        ESTALE           = 0x1004B,           // Reserved.
+        ETIME            = 0x1004C,           // Stream ioctl() timeout.
+        ETIMEDOUT        = 0x1004D,           // Connection timed out.
+        ETXTBSY          = 0x1004E,           // Text file busy.
+        EXDEV            = 0x1004F,           // Cross-device link.
+
+        // POSIX permits these to have the same value and we make them always equal so
+        // that CoreFX cannot introduce a dependency on distinguishing between them that
+        // would not work on all platforms.
+        EOPNOTSUPP      = ENOTSUP,            // Operation not supported on socket
+        EWOULDBLOCK     = EAGAIN,             // Operation would block
+    }
+
+    // Represents a platform-agnostic Error and underlying platform-specific errno
+    internal struct ErrorInfo
+    {
+        internal readonly Error Error;
+        internal readonly int Errno;
+
+        internal ErrorInfo(Error error)
+        {
+            Error = error;
+            Errno = Interop.System.ConvertErrorPalToPlatform(error);
```

Comment:
I'll file an issue to investigate, but stick to the current approach for the initial commit. You're right that it could be shared that way -- I hadn't thought of it. It is slightly tricky because we'd need to share something that is efficient to lookup in directly (whereas I was imagining building the lookup from an unordered list in managed code). The Pal -> Platform case can just be an index into an array so that's easy. However, the more interesting and common Platform -> Pal case can be sparse. I'd be inclined to use a sorted array of (int, Error) + binary search to keep the shared data structure trivial. It'll take me some time to measure, etc., and it's just complex enough to warrant its own review IMHO.